### PR TITLE
Move dnsmasq image build from kubernetes/contrib to kubernetes/dns

### PR DESCRIPTION
**Pull request by bowei**

### CODE REVIEW

**Comment by MrHohn** [comment]
cc @spxtr seems like Travis is still not working?

**Comment by MrHohn** [comment]
As dnsmasq is put in the image folder and has its own Makefile, we will need to make and push it separately. This seems a bit confusing, because we may want to manage all dns container images in the same manner.

Is there anyway to combine them together?

File: README.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ This is the repository for [Kubernetes DNS](http://kubernetes.io/docs/admin/dns/
 ## Subprojects
 
 * [sidecar](docs/sidecar/README.md)
+* [dnsmasq container](images/dnsmasq/RELEASES.md)
```
Comment (MrHohn) [inline README.md]: Probably create a short README.md for dnsmasq and link it here? RELEASE.md does not mention what this dnsmasq is...

File: images/dnsmasq/Changelog [hunk 2]
```diff
@@ -0,0 +1,8 @@
+## Version 1.1 (Fri May 27 2016 Abhi S. <abshah@google.com>
+ - First AMD64-only dnsmasq container
+## Version 1.2 (Wed June 8 2016 Girish Kalele <gkalele@google.com>
+ - Multiple Architecture builds and upgraded to Alpine 3.4
+## Version 1.3 (Fri June 11 2016 Lucas Käldström <lucas.kaldstrom@hotmail.co.uk>)
+ - Patched the multi-architecture images, so they'll work
+## Version 1.4 (Thu September 29 2016 Zihong Zheng <zihongz@google.com>)
+ - Redirect STOPSIGNAL for graceful termination
```
Comment (MrHohn) [inline images/dnsmasq/Changelog]: I also wonder how should we manage changelogs from different dns components. If we both agree on unified versioning, should we maintain only one changelog afterward?

Though keep changelogs separate may make things clearer.

**Comment by spxtr** [comment]
Looks like it's okay. Travis has been flaking for us lately, they might be having some issues.

**Comment by MrHohn** [comment]
@spxtr Yeah it is working now, thanks!

**Comment by bowei** [comment]
@spxtr @MrHohn -- travis was a permission issue, fixed (thanks to thockin)

**Comment by bowei** [comment]
I will integrate the builds -- the dnsmasq build needs to be improved to work better

**Pull request closed without merging**